app/controllers/topics_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TopicsController < ApplicationController
+  def index
+    @topics = Topic.all
+  end
+
+  def show
+    @topic = Topic.find(params[:id])
+  end
+
+  def new
+    @user = current_user
+    @topic = Topic.new
+  end
+  
+  def create
+    @user = current_user
+    @topic = @user.topics.build(topic_params)
+    
+    if @topic.save
+      redirect_to(topics_path)
+    else
+      flash[:alert] = 'There was an error creating the topic'
+      redirect_to(new_topic)
+    end
+  end
+
+  def edit
+    @topic = Topic.find(params[:id])
+  end
+  
+  def destroy
+    @user = current_user
+    @topic = @user.topics.find(params[:id])
```

Comment:
@ianroberts131 👍 good protection of `@topic` with the `@user`-scoped query. Consider how the other actions in this controller might be vulnerable to exposing topic data to attackers. How can those actions protect the topic data?